### PR TITLE
update the aggregation value based on final compute

### PR DESCRIPTION
**Pull request by sreemanamala**
### Description

This PR tries to fix the aggregated result in the window frame by invoking the finalize computation of corresponding aggregator factory prior to storing the result.

<hr>

##### Key changed/added classes in this PR
 * `DefaultFramedOnHeapAggregatable.java`

<hr>

This PR has:

- [ ] been self-reviewed.
   - [ ] using the [concurrency checklist](https://github.com/apache/druid/blob/master/dev/code-review/concurrency.md) (Remove this item if the PR doesn't have any relation to concurrency.)
- [ ] added documentation for new or modified features or behaviors.
- [ ] a release note entry in the PR description.
- [ ] added Javadocs for most classes and all non-trivial methods. Linked related entities via Javadoc links.
- [ ] added or updated version, license, or notice information in [licenses.yaml](https://github.com/apache/druid/blob/master/dev/license.md)
- [ ] added comments explaining the "why" and the intent of the code wherever would not be obvious for an unfamiliar reader.
- [ ] added unit tests or modified existing tests to cover new code paths, ensuring the threshold for [code coverage](https://github.com/apache/druid/blob/master/dev/code-review/code-coverage.md) is met.
- [ ] added integration tests.
- [ ] been tested in a test Druid cluster.

### CODE REVIEW

File: processing/src/main/java/org/apache/druid/query/rowsandcols/semantic/DefaultFramedOnHeapAggregatable.java [hunk 1]
```diff
@@ -321,7 +321,7 @@ public static Object cloneAggValue(AggregatorFactory aggFactory, Object value)
     Aggregator combiningAgg = aggFactory.getCombiningFactory().factorize(combiningFactory);
 
     combiningAgg.aggregate();
-    return combiningAgg.get();
```
Comment (kgyrtkirk) [inline processing/src/main/java/org/apache/druid/query/rowsandcols/semantic/DefaultFramedOnHeapAggregatable.java:324]: note: I think we miss `combiningAgg.close()` here - not sure if hat matters...but could be fixed with a try-with-resources

File: processing/src/main/java/org/apache/druid/query/rowsandcols/semantic/DefaultFramedOnHeapAggregatable.java [hunk 2]
```diff
@@ -719,7 +719,7 @@ private AppendableRowsAndColumns aggregateWindowApertureWellBehaved(
     // End Phase 3, anything left in the window needs to be collected and put into our results
     for (; nextIndex < windowSize; ++nextIndex) {
       for (int i = 0; i < aggFactories.length; ++i) {
-        results[i][resultStorageIndex] = aggregators[i][nextIndex].get();
```
Comment (kgyrtkirk) [inline processing/src/main/java/org/apache/druid/query/rowsandcols/semantic/DefaultFramedOnHeapAggregatable.java:722]: is `aggregators[i][nextIndex].close();` missing here?

**Review by kgyrtkirk — APPROVED** [review]

File: processing/src/main/java/org/apache/druid/query/rowsandcols/semantic/DefaultFramedOnHeapAggregatable.java (hunk 2 above)
Comment (sreemanamala) [reply processing/src/main/java/org/apache/druid/query/rowsandcols/semantic/DefaultFramedOnHeapAggregatable.java:722]: yes, added it

File: processing/src/main/java/org/apache/druid/query/rowsandcols/semantic/DefaultFramedOnHeapAggregatable.java (hunk 1 above)
Comment (sreemanamala) [reply processing/src/main/java/org/apache/druid/query/rowsandcols/semantic/DefaultFramedOnHeapAggregatable.java:324]: added it

**Pull request merged**